src/pages/[platform]/build-a-backend/more-features/in-app-messaging/in-app-messaging-migration-guide/index.mdx

```diff
@@ -39,138 +39,545 @@ export function getStaticProps(context) {
-+ onMessageDismissed(myMessageDismissedHandler);
+    listener.remove(); // Remember to remove the listener when it is no longer needed
+    ```
 
-- InAppMessaging.onMessageActionTaken(myMessageActionTakenHandler);
-+ onMessageActionTaken(myMessageActionTakenHandler);
-```
+  </Block>
+</BlockSwitcher>
+
+### InAppMessaging.onMessageDismissed
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { onMessageDismissed } from 'aws-amplify/in-app-messaging';
+
+    const myMessageDismissedHandler = (message) => {
+      // Do something with the dismissed message
+    };
+
+    const listener = onMessageDismissed(myMessageDismissedHandler);
+
+    listener.remove(); // Remember to remove the listener when it is no longer needed
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Notifications } from 'aws-amplify';
+    const { InAppMessaging } = Notifications;
+
+    const myMessageDismissedHandler = (message) => {
+      // Do something with the dismissed message
+    };
+
+    const listener = InAppMessaging.onMessageDismissed(myMessageDismissedHandler);
+
+    listener.remove(); // Remember to remove the listener when it is no longer needed
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### InAppMessaging.onMessageActionTaken
+
+<BlockSwitcher>
+  <Block name="V6">
+    ```ts
+    import { onMessageActionTaken } from 'aws-amplify/in-app-messaging';
+
+    const myMessageActionTakenHandler = (message) => {
+      // Do something with the message action was taken against
+    };
+
+    const listener = onMessageActionTaken(myMessageActionTakenHandler);
+
+    listener.remove(); // Remember to remove the listener when it is no longer needed
+    ```
+
+  </Block>
+  <Block name="V5">
+    ```ts
+    import { Notifications } from 'aws-amplify';
+    const { InAppMessaging } = Notifications;
+
+    const myMessageActionTakenHandler = (message) => {
+      // Do something with the message action was taken against
+    };
+
+    const listener = InAppMessaging.onMessageActionTaken(
+      myMessageActionTakenHandler
+    );
+
+    listener.remove(); // Remember to remove the listener when it is no longer needed
+    ```
+
+  </Block>
+</BlockSwitcher>
+
+### InAppMessaging.notifyMessageInteraction
+
+The way to notify any interaction event listeners of your users' interactions with in-app messages in v6 has not changed in behavior. You will continue to use the `notifyMessageInteraction` API to notify of interaction events. However, this API is now exported from the `aws-amplify/in-app-messaging` path and the input parameters have been updated as follows:
+
+- Instead of two position positional parameters (corresponding to `message` and `inAppMessageInteractionEvent`), there are now two named parameters: `message` and `type` (i.e. a single input object containing these properties).
+- Interaction event types have been changed from a SCREAMING_SNAKE_CASE convention to camelCase.
```

Comment:
`SCREAMING_SNAKE_CASE` is the actual name for it?

Reply:
Yeah 😅